src/geo/transform.js

```diff
@@ -1261,11 +1254,21 @@ class Transform {
     }
 
     calculatePosMatrix(unwrappedTileID: UnwrappedTileID, worldSize: number): Float32Array {
-        const posMatrix = mat4.identity(new Float64Array(16));
+        const isMercator = this.projection.name === 'mercator';
         const cs = this.projection.tileTransform(unwrappedTileID.canonical);
-        mat4.scale(posMatrix, posMatrix, [1 /  cs.scale, 1 /  cs.scale, 1]);
-        mat4.translate(posMatrix, posMatrix, [cs.x, cs.y, 0]);
-        mat4.scale(posMatrix, posMatrix, [1 / EXTENT, 1 / EXTENT, 1]);
+        const canonical = unwrappedTileID.canonical;
+        const scale = isMercator ? worldSize / this.zoomScale(canonical.z) : 1;
+        const unwrappedX = canonical.x + Math.pow(2, canonical.z) * unwrappedTileID.wrap;
+        const scaledX = isMercator ? unwrappedX * scale : cs.x;
+        const scaledY = isMercator ? canonical.y * scale : cs.y;
+
+        const posMatrix = mat4.identity(new Float64Array(16));
+        if (!isMercator) {
+            mat4.scale(posMatrix, posMatrix, [scale / cs.scale, scale / cs.scale, 1]);
```

Comment:
Nit: ideally we would clean this up so that we only generate `tileTransform` if it's not mercator. There are also a lot of `isMercator` ternaries that would look better if refactored into a single `isMercator` `if` block.